.github/workflows/publish.yml

```diff
@@ -0,0 +1,54 @@
+name: Publish to PyPi
+
+# only build & publish to PyPi once release notes are published
+# this ensures we have release notes correctly, tags, changelog, etc.
+
+on:
+  release:
+    types: [published] 
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: "3.8"
+    - name: Set release notes tag
+      run: |
+        export RELEASE_TAG_VERSION=${{ github.event.release.tag_name }}
+        echo ::set-env name=RELEASE_TAG_VERSION::${RELEASE_TAG_VERSION:1}
+    - name: Ensure new version is also set in pyproject and CHANGELOG
+      run: |
+        grep ${RELEASE_TAG_VERSION} CHANGELOG.md
+        grep ${RELEASE_TAG_VERSION} pyproject.toml
```

Comment:
This could accidentally match against a dependency version, rather than the project's version. Should we search for `version = "${RELEASE_TAG_VERSION}"` instead?

Reply:
Good catch - I've updated it both searches with exact match now:

```shell
        grep --regexp "\[${RELEASE_TAG_VERSION}\]" CHANGELOG.md
        grep --regexp "version \= \"${RELEASE_TAG_VERSION}\"" pyproject.toml
```

As I can't be sure whether Github Actions will have a GNU grep or not, I'm using a basic regex